### ISSUE TITLE
Changing the tf_random_seed (RunConfig) doesn't change results/parameters for LinearClassifier

### ISSUE THREAD

**Issue by echo66** [tensorflow/tensorflow#27896]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04.2 LTS
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): N/A
- TensorFlow version (use command below): v1.13.1-0-g6612da8951
- Python version: 3.6.7
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the current behavior**
Using different `tf_random_seed` values for `tf.estimator.RunConfig` produces the same model performance and parameters for `tf.estimator.LinearClassifier`.

**Describe the expected behavior**
Different seeds to produce different performance/parameters, specially if the dataset is small or if the optimizer only takes a step for a small batch size.

**Code to reproduce the issue**
```python
import pandas as pd
import numpy as np

from sklearn.datasets import make_classification
from sklearn.metrics import roc_auc_score

import tensorflow as tf
from tensorflow.logging import set_verbosity
from tensorflow.estimator.inputs import pandas_input_fn
from tensorflow.estimator import LinearClassifier
from tensorflow.feature_column import numeric_column


X, y = make_classification(n_samples=50000, n_classes=2, 
                           n_clusters_per_class=4, class_sep=1.0, 
                           random_state=10, weights=[0.8, 0.2], 
                           n_features=10, n_informative=5)

X = pd.DataFrame(X, columns=["f{}".format(i) for i in range(10)])
y = pd.Series(y)

input_fn = pandas_input_fn(X, y=y, batch_size=5, num_epochs=1, shuffle=False)

feature_columns = [numeric_column(key=c) for c in X.columns]

classifier = LinearClassifier(feature_columns, config=tf.estimator.RunConfig(tf_random_seed=10))

classifier.train(input_fn, steps=1)

print('\n\n\n')

for c in X.columns:
    print(c, classifier.get_variable_value("linear/linear_model/{}/weights".format(c))[0][0])

feature_columns = [numeric_column(key=c) for c in X.columns]

classifier = LinearClassifier(feature_columns, config=tf.estimator.RunConfig(tf_random_seed=31231))

classifier.train(input_fn, steps=1)

print('\n\n\n')

for c in X.columns:
    print(c, classifier.get_variable_value("linear/linear_model/{}/weights".format(c))[0][0])
```

**Other info / logs**
```
f0 -0.1945913
f1 -0.19702944
f2 0.1967695
f3 0.19892408
f4 -0.18479808
f5 -0.19904605
f6 0.1991331
f7 -0.19970222
f8 0.186639
f9 -0.1857145




f0 -0.1945913
f1 -0.19702944
f2 0.1967695
f3 0.19892408
f4 -0.18479808
f5 -0.19904605
f6 0.1991331
f7 -0.19970222
f8 0.186639
f9 -0.1857145
```

**Comment by jvishnuvardhan** [comment]
@echo66 I think as you set `shuffle=False`, the input data is same for both the cases and hence the model parameters are same. I tried 'shuffle=True' and removed 'tf_random_seed'. Now the parameters are different. Thanks!

```
import pandas as pd
import numpy as np

from sklearn.datasets import make_classification
from sklearn.metrics import roc_auc_score

import tensorflow as tf
from tensorflow.logging import set_verbosity
from tensorflow.estimator.inputs import pandas_input_fn
from tensorflow.estimator import LinearClassifier
from tensorflow.feature_column import numeric_column


X, y = make_classification(n_samples=50000, n_classes=2, 
                           n_clusters_per_class=4, class_sep=1.0, 
                           random_state=10, weights=[0.8, 0.2], 
                           n_features=10, n_informative=5)

X = pd.DataFrame(X, columns=["f{}".format(i) for i in range(10)])
y = pd.Series(y)

input_fn = pandas_input_fn(X, y=y, batch_size=5, num_epochs=1, shuffle=True)

feature_columns = [numeric_column(key=c) for c in X.columns]

classifier = LinearClassifier(feature_columns, config=tf.estimator.RunConfig())

classifier.train(input_fn, steps=1)

print('\n\n\n')

for c in X.columns:
    print(c, classifier.get_variable_value("linear/linear_model/{}/weights".format(c))[0][0])

feature_columns = [numeric_column(key=c) for c in X.columns]

classifier = LinearClassifier(feature_columns, config=tf.estimator.RunConfig())

classifier.train(input_fn, steps=1)

print('\n\n\n')

for c in X.columns:
    print(c, classifier.get_variable_value("linear/linear_model/{}/weights".format(c))[0][0])
```

**Comment by echo66** [comment]
Hello @jvishnuvardhan !

In the code snippet, you have, at least, two processes that (should) require seeds for random numbers generators: (a) dataset shuffling and (b) linear model parameters initialization. If I change the seed, I still get the same parameters' values at the end of a single short optimization phase. This shouldn't be the case, unless (1) the parameters' initialization doesn't depend on the seed or (2) there is some kind of closed form solution for the model (AFAIK, FTRL optimizer is iterative).

**Comment by uroosehar1** [comment]
Yeah may be we have some basic concept missed about Lnear classifier
I got the basic idea about Linear classifiers from one tutorial 
here is the link for this
URL: https://youtu.be/PKhojDaOi9Q

**Comment by bhack** [comment]
Ok the issue is that in the `LinearClassifier` the inits are `tf.keras.initializers.zeros()` so we don't have any seed effects:
https://github.com/tensorflow/estimator/blob/2248336960a2cc65d2b7d0b21aca76157ccd0e79/tensorflow_estimator/python/estimator/canned/linear.py#L1464-L1472 

/cc @qlzh727 What we want to do?

**Comment by sushreebarsa** [comment]
@echo66 Could you please try on latest stable version of tf 2.5 or 2.4.1 and let us know if this is still an issue.Thanks!
